OpenRA.Mods.Common/Traits/BotModules/BotModuleLogic/BaseBuilderQueueManager.cs

```diff
@@ -226,6 +231,20 @@ ActorInfo ChooseBuildingToBuild(ProductionQueue queue)
 		{
 			var buildableThings = queue.BuildableItems();
 
+			if (baseBuilder.Info.WalledStructures.Count > 0)
+			{
+				var wall = GetProducibleBuilding(baseBuilder.Info.WallTypes, buildableThings);
+				if (wall != null && HasSufficientPowerForActor(wall))
+				{
+					var possibleLocation = ChooseWallLocation(wall);
+					if (possibleLocation != null)
+					{
+						AIUtils.BotDebug($"{queue.Actor.Owner} decided to build {wall.Name}: Priority override (wall)");
+						return wall;
+					}
+				}
+			}
```

Comment:
I feel like this should be the last override to consider, power and resources seem more important than walls.

Reply:
It runs in a separate queue, so it doesn't block the other ones. Also, I need this as early as possible for best success as I don't have https://github.com/OpenRA/OpenRA/pull/19513 yet and there is a chance the AI might block the complete walling with own structures.

Reply:
> It runs in a separate queue
Not in D2k, and doesn't have to be the case in 3rd party mods either (like RA Classic).

Reply:
Moved it.